service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/ainotebook/post-startup.sh

```diff
@@ -756,6 +756,32 @@ fi
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASHRC}"
 chown ${LOGIN_USER}:${LOGIN_USER} "${USER_BASH_PROFILE}"
 
+#######################################
+# Restart proxy to pick up the new env
+######################################
+APP_PROXY=$(get_metadata_value "instance/attributes/terra-app-proxy")
+readonly PROXY_ENV="/opt/deeplearning/proxy.env"
+if [[ -n "${APP_PROXY}" ]]; then
+    echo "Using custom Proxy Agent"
+    RESOURCE_ID=$(get_metadata_value "instance/attributes/terra-resource-id")
+    NEW_PROXY="https://${APP_PROXY}"
+    ESCAPED_NEW_PROXY=$(sed 's/[\/&]/\\&/g' <<< "${NEW_PROXY}")
```

Comment:
Can you add some comments about this block. It's not obvious from reading what the various bits of string manipulation are intended to achieve.